From 1f5e6f22defaaa106643f00caabe74a4f3da1d01 Mon Sep 17 00:00:00 2001
From: David Morel <david.morel@vates.fr>
Date: Mon, 31 Jul 2023 17:47:01 +0200
Subject: [PATCH 25/32] [W] Patch76:
 openssl-1.1.1-cleanup-peer-point-reneg.patch

1 fewer insertion, 1 fewer deletion:

Original:
 extensions.c |   12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)
Staged:
 extensions.c |   10 ++++++++++
 1 file changed, 10 insertions(+)
---
 ssl/statem/extensions.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/ssl/statem/extensions.c b/ssl/statem/extensions.c
index 0f39275baa..af3ac4d950 100644
--- a/ssl/statem/extensions.c
+++ b/ssl/statem/extensions.c
@@ -43,6 +43,7 @@ static int tls_parse_certificate_authorities(SSL *s, PACKET *pkt,
 #ifndef OPENSSL_NO_SRP
 static int init_srp(SSL *s, unsigned int context);
 #endif
+static int init_ec_point_formats(SSL *s, unsigned int context);
 static int init_etm(SSL *s, unsigned int context);
 static int init_ems(SSL *s, unsigned int context);
 static int final_ems(SSL *s, unsigned int context, int sent);
@@ -1175,6 +1176,15 @@ static int init_srp(SSL *s, unsigned int context)
 }
 #endif
 
+static int init_ec_point_formats(SSL *s, unsigned int context)
+{
+	    OPENSSL_free(s->ext.peer_ecpointformats);
+	    s->ext.peer_ecpointformats = NULL;
+	    s->ext.peer_ecpointformats_len = 0;
+
+	    return 1;
+}
+
 static int init_etm(SSL *s, unsigned int context)
 {
     s->ext.use_etm = 0;
-- 
2.41.0

